### PR TITLE
feat: add kill switch abstraction

### PR DESCRIPTION
**Pull request by Marc-Antoine-Soucy**
GitHub Issue: #

## Proposed Changes

 - [ ] Bug fix
 - [X] Feature
 - [ ] Code style update (formatting)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build or CI related changes
 - [ ] Documentation content changes
 - [ ] Other, please describe:

## Description

Added the killswitch

## Impact on version

- [ ] **Major**
  - The template structure was changed.
- [X] **Minor**
  - New functionalities were added.
- [ ] **Patch**
  - A bug in behavior was fixed.
  - Documentation was changed.

### Always applicable
No matter your changes, these checks always apply.
- [X] Your conventional commits are aligned with the **Impact on version** section.
- [X] Updated [CHANGELOG.md](../CHANGELOG.md).
  - Use the latest Major.Minor.X header if you do a **Patch** change.
  - Create a new Major.Minor.X header if you do a **Minor** or **Major** change.
  - If you create a new header, it aligns with the **Impact on version** section and matches what is generated in the build pipeline.
- [X] Documentation files were updated according with the changes.
- [X] Images about the template are referenced from the [wiki](https://github.com/nventive/UnoApplicationTemplate/wiki/Images) and added as images in this git.
- [X] TODO comments are hints for next steps for users of the template and not planned work.

### Contextual
Based on your changes these checks may not apply.
- [ ] Automated tests for the changes have been added/updated.
- [x] Tested on all relevant platforms

### CODE REVIEW

File: doc/Architecture.md [hunk 1]
```diff
@@ -72,6 +72,11 @@ This application contains a forced update feature.
 
 See [ForcedUpdate.md](ForcedUpdate.md) for more details.
 
+### Kill Switch
+This application contains Kill switch feature.
```
Comment (Soap-141) [inline doc/Architecture.md:76]: Add blank line here between content and title.

File: src/app/lib/presentation/diagnostic/navigation_diagnostic_widget.dart [hunk 2]
```diff
@@ -12,6 +14,7 @@ final class NavigationDiagnosticWidget extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final minimumVersionRepository = GetIt.I.get<MinimumVersionRepository>();
+    final killSwitchRepository = GetIt.I.get<KillSwitchRepository>();
     return Row(
```
Comment (Soap-141) [inline src/app/lib/presentation/diagnostic/navigation_diagnostic_widget.dart:18]: Add a blank line between return and variables, will make it cleaner.

**Review by Soap-141 — COMMENTED** [review]
Good job! I see that you added Rx Dart!

File: doc/KillSwitch.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Kill Switch
+
+The kill switch feature is for when you want to temporarily lock the user out of the app. 
+This could be used for example when the server is down for some time, to avoid the users getting a ton of errors and getting reports from those users.
+
+To trigger the kill switch, we suscribe to the `isKillSwitchActivatedStream` `stream` from the `KillSwitchService` in the [Main](../src/app/lib/main.dart).
```
Comment (jeanplevesque) [inline doc/KillSwitch.md:6]: subscribe

**Review by jeanplevesque — APPROVED** [review]

**Review by Soap-141 — APPROVED** [review]

**Pull request merged**